src/query/stream/stream_consumer_factory.cpp

```diff
@@ -0,0 +1,237 @@
+    MG_ASSERT(result.signature->contains(query_param_name));
+    MG_ASSERT(result.signature->contains(params_param_name));
+
+    spdlog::trace("Calling transformation in stream '{}'", stream_name);
+    trans.cb(&mgp_messages, &graph, &result, &memory);
+  }
+  if (result.error_msg.has_value()) {
+    throw StreamsException(result.error_msg->c_str());
+  }
+}
+}  // namespace
+
+template <typename TStream>
+auto make_consumer_impl(memgraph::dbms::DatabaseAccess db_acc, const std::string &stream_name,
+                        const std::string &transformation_name, std::optional<std::string> owner,
+                        memgraph::query::InterpreterContext *interpreter_context)
+    -> IStreamConsumerFactory::Consumer<TStream> {
+  auto interpreter = std::make_shared<memgraph::query::Interpreter>(interpreter_context, std::move(db_acc));
+
+  auto *memory_resource = memgraph::utils::NewDeleteResource();
+  // there is an advantage of with the result being reused (retains internal buffers)
+  auto result = mgp_result{nullptr, memory_resource};
+
+  return [result = std::move(result), memory_resource, interpreter_context, interpreter = std::move(interpreter),
+          stream_name, transformation_name,
+          owner = std::move(owner)](const std::vector<typename TStream::Message> &messages) mutable {
+#ifdef MG_ENTERPRISE
+    interpreter->OnChangeCB([](auto) { return false; });  // Disable database change
+#endif
+    auto accessor = interpreter->current_db_.db_acc_->get()->Access();
+    // register new interpreter into interpreter_context
+    interpreter_context->interpreters->insert(interpreter.get());
+    memgraph::utils::OnScopeExit interpreter_cleanup{
+        [interpreter_context, interpreter]() { interpreter_context->interpreters->erase(interpreter.get()); }};
+
+    memgraph::metrics::IncrementCounter(memgraph::metrics::MessagesConsumed, messages.size());
+
+    CallCustomTransformation(transformation_name, messages, result, *accessor, *memory_resource, stream_name);
+
+    memgraph::query::DiscardValueResultStream stream;
+
+    spdlog::trace("Start transaction in stream '{}'", stream_name);
+    memgraph::utils::OnScopeExit cleanup{[&interpreter, &result]() {
+      result.rows.clear();
+      interpreter->Abort();
+    }};
+
+    uint32_t i = 0;
+    auto const total_retries = interpreter_context->config.stream_transaction_conflict_retries;
+    const auto retry_interval = interpreter_context->config.stream_transaction_retry_interval;
+    while (true) {
+      try {
+        interpreter->BeginTransaction();
+        for (auto &row : result.rows) {
+          spdlog::trace("Processing row in stream '{}'", stream_name);
+          auto [query_value, params_value] = ExtractTransformationResult(row.values, transformation_name, stream_name);
+          memgraph::storage::PropertyValue params_prop{params_value};
+
+          std::string query{query_value.ValueString()};
```

Comment:
```suggestion
          const std::&string query{query_value.ValueString()};
```

I think no need to copy it here

Reply:
Going from pmr::string to string